src/lib/network_pool/snark_pool.ml

```diff
@@ -481,91 +490,101 @@ let%test_module "random set test" =
 
     let%test_unit "Work that gets fed into apply_and_broadcast will be \
                    received in the pool's reader" =
-      let pool_reader, _pool_writer = Linear_pipe.create () in
-      let frontier_broadcast_pipe_r, _ =
-        Broadcast_pipe.create (Some (Mocks.Transition_frontier.create ()))
-      in
-      let network_pool =
-        Mock_snark_pool.create ~logger:(Logger.null ())
-          ~pids:(Child_processes.Termination.create_pid_set ())
-          ~trust_system ~incoming_diffs:pool_reader
-          ~frontier_broadcast_pipe:frontier_broadcast_pipe_r
-      in
-      let priced_proof =
-        { Priced_proof.proof=
-            `One
-              (mk_dummy_proof
-                 (Quickcheck.random_value ~seed:(`Deterministic "test proof")
-                    Transaction_snark.Statement.gen))
-        ; fee=
-            { fee= Currency.Fee.of_int 0
-            ; prover= Signature_lib.Public_key.Compressed.empty } }
-      in
-      let command =
-        Mock_snark_pool.Resource_pool.Diff.Stable.V1.Add_solved_work
-          (fake_work, priced_proof)
-      in
-      (fun () ->
-        don't_wait_for
-        @@ Linear_pipe.iter (Mock_snark_pool.broadcasts network_pool)
-             ~f:(fun _ ->
-               let pool = Mock_snark_pool.resource_pool network_pool in
-               ( match
-                   Mock_snark_pool.Resource_pool.request_proof pool fake_work
-                 with
-               | Some {proof; fee= _} ->
-                   assert (proof = priced_proof.proof)
-               | None ->
-                   failwith "There should have been a proof here" ) ;
-               Deferred.unit ) ;
-        Mock_snark_pool.apply_and_broadcast network_pool
-          (Envelope.Incoming.local command) )
-      |> Async.Thread_safe.block_on_async_exn
+      Async.Thread_safe.block_on_async_exn (fun () ->
```

Comment:
sad that you have to wrap these in block_on_async now, but we only have one bind in these so it shouldn't slow down tests too much